File: selection.c
// complete the selection_sort function

#include <cs50.h>
#include <stdio.h>

// function prototypes
void selection_sort(int arr[], int size);
void print_array(int arr[], int size);

// size of array
#define SIZE 10

int main(void)
{ 
    // initialize array
    int arr[] = {1, 8, 4, 6, 0, 3, 5, 2, 7, 9};
  
    // sort array
    selection_sort(arr, SIZE); 
    
    // print out the array
    print_array(arr, SIZE); 
    
    // done
    return 0; 
} 

// Function to print an array
void print_array(int arr[], int size) 
{ 
    for (int i = 0; i < size; i++) 
    {
        printf("%i ", arr[i]); 
    }
    printf("\n"); 
}

// selection sort
void selection_sort(int arr[], int size) 
{
    // TODO
    int s = 0;
    int a = 0;
    for (int i = 0; i < size - 1; i++)
    {
        s = i;
        for (int j = i + 1; j < size - 1; j++)
        {
            if (arr[j] < arr[s])
            {
                s = j;
            }
        }
        a = arr[s];
        arr[s] = arr[i];
        arr[i] = a;
    }
}

File: selection.py
# Complete the selection sort function

# List of numbers to sort
numbers = [1, 8, 4, 6, 0, 3, 5, 2, 7, 9]


def main():
    selection_sort(numbers)

    for number in numbers:
        print(number, end=" ")
    print()


# Sort list using selection sort
def selection_sort(numbers):
    s = 0
    a = 0
    n = len(numbers)
    for i in range(n - 1):
        s = i
        for j in range(i + 1, n - 1):
            if numbers[j] < numbers[s]:
                s = j
        a = numbers[s]
        numbers[s] = numbers[i]
        numbers[i] = a
    return


main()
